test/integration/examples/apiserver_test.go

```diff
@@ -225,164 +232,143 @@ func TestAPIServiceWaitOnStart(t *testing.T) {
 	}
 }
 
-func TestAggregatedAPIServer(t *testing.T) {
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (f roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+// TestFrontProxyConfig tests that the RequestHeader configuration is consumed
+// correctly by the aggregated API servers.
+func TestFrontProxyConfig(t *testing.T) {
+	const testNamespace = "integration-test-front-proxy-config"
+
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
 	t.Cleanup(cancel)
 
-	// makes the kube-apiserver very responsive.  it's normally a minute
-	dynamiccertificates.FileRefreshDuration = 1 * time.Second
+	// start up the KAS and prepare the options for the wardle API server
+	testKAS, wardleOptions, wardlePort := prepareAggregatedWardleAPIServer(ctx, t, testNamespace, []string{"x-remote-uid"})
```

Comment:
Also add a test case where the UID header field is not set and thus the SA token UID is not sent to the aggregated API server.  In the future, this test will fail when we handle the defaulting logic in a way to separate KAS front proxy config vs. aggregated API server front proxy config